### DISCUSSION TITLE
'pyvista' has no attribute 'POpenFOAMReader'

### DISCUSSION THREAD

**Discussion by govindiitd** [pyvista/pyvista#5264] [Q&A]
Hi,

I am at early stage with pyvista but not with python. I have openfoam simulation results want to explore pyvista for visualization. I have installed pyvista with conda on windows but POpenFOAMReader seems to be no where close. 

Any suggestion will be appreciated.

Regards,
Govind

**Comment by MatthewFlamm** [comment]
What version of PyVista are you using?  Please share `pv.Report()`

**Comment by govindiitd** [comment]
Yes, I upgraded the version, it works fine now.

I am practicing with an OpenFOAM cavity example of mine, not a downloaded
one. I get the following error:

'Data array (U) not present in this dataset.'


However, I checked that its present.


Cell Data:
pyvista DataSetAttributes
Association     : CELL
Active Scalars  : p
Active Vectors  : U
Active Texture  : None
Active Normals  : None
Contains arrays :
    U                       float32    (400, 3)             VECTORS
    p                       float32    (400,)               SCALARS

Point Data:
pyvista DataSetAttributes
Association     : POINT
Active Scalars  : p
Active Vectors  : U
Active Texture  : None
Active Normals  : None
Contains arrays :
    U                       float32    (882, 3)             VECTORS
    p                       float32    (882,)               SCALARS




I am not able to figure out why this?


Regards,

Govind

**Reply by MatthewFlamm** [reply]
Can you share the code you are using?

**Comment by govindiitd** [comment]
Please find the attached notebook; however, its the same given by pyvista
openfoam tutorial except that I am reading the custom results run on my
system.

Together, I have attached the case setup also.



Govind

**Reply by MatthewFlamm** [reply]
I do not see an attachment here.  Are you sure that your data is exactly like the example data (we also have more than one openFOAM example).  If you are simply running the cavity tutorial with a different cavity case, then can you point to the line that gives you the error?  Are you sure that your data is the same as the example?  Same location, same size, same variable data?

**Reply by govindiitd** [reply]
[cavity.zip](https://github.com/pyvista/pyvista/files/13575908/cavity.zip)
 
This is the case setup with jupyter-notebook.

**Comment by govindiitd** [comment]
I resolved this one. It was my error only; I missed to set
"cell_to_point_creation = False".


Thanks,
Govind
